In [237]:
#Importing Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import MinMaxScaler


In [238]:
#Load the dataset and display first 5 datas
diab_df = pd.read_csv("diabetes.csv") 
diab_df.head()

age       sex       bmi        bp        s1        s2        s3  \
0  0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401   
1 -0.001882 -0.044642 -0.051474 -0.026328 -0.008449 -0.019163  0.074412   
2  0.085299  0.050680  0.044451 -0.005670 -0.045599 -0.034194 -0.032356   
3 -0.089063 -0.044642 -0.011595 -0.036656  0.012191  0.024991 -0.036038   
4  0.005383 -0.044642 -0.036385  0.021872  0.003935  0.015596  0.008142   

         s4        s5        s6  Progression  
0 -0.002592  0.019907 -0.017646        151.0  
1 -0.039493 -0.068332 -0.092204         75.0  
2 -0.002592  0.002861 -0.025930        141.0  
3  0.034309  0.022688 -0.009362        206.0  
4 -0.002592 -0.031988 -0.046641        135.0

In [239]:
#Checking for null values
diab_df.shape

(442, 11)

In [240]:
#Getting independent variable in X.
x = diab_df.iloc[:,[0,1,2,3,4,5,6,7,9]].values
x

array([[ 0.03807591,  0.05068012,  0.06169621, ..., -0.04340085,
        -0.00259226, -0.01764613],
       [-0.00188202, -0.04464164, -0.05147406, ...,  0.07441156,
        -0.03949338, -0.09220405],
       [ 0.08529891,  0.05068012,  0.04445121, ..., -0.03235593,
        -0.00259226, -0.02593034],
       ...,
       [ 0.04170844,  0.05068012, -0.01590626, ..., -0.02499266,
        -0.01107952,  0.01549073],
       [-0.04547248, -0.04464164,  0.03906215, ..., -0.02867429,
         0.02655962, -0.02593034],
       [-0.04547248, -0.04464164, -0.0730303 , ...,  0.17381578,
        -0.03949338,  0.00306441]])

In [241]:
#Getting dependent variable in y.

y = diab_df.iloc[:,10].values
y = y.reshape(-1, 1)
X = x.reshape(-1, 9)


### Training and test data

In [242]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2,shuffle=True)

print ("Training data:",X_train.shape, y_train.shape)
print ("Test data:",X_test.shape, y_test.shape)

Training data: (353, 9) (353, 1)
Test data: (89, 9) (89, 1)


Using a 0.2 test size split our data of 442 items into a training set of 89 observations and nine columns, and a test set of 89 observations.


### MinMaxScaler and StandardScaler

In [243]:
#Fit the scaler on train data
ms = MinMaxScaler()
ms.fit(X_train)


# Apply the scaler on train and test data

X_train = ms.transform(X_train)
X_test = ms.transform(X_test)


In [244]:
#import StandardScaler
from sklearn.preprocessing import StandardScaler

In [245]:
#Create the object
std_scaler = StandardScaler()
#fit and transform the training set 
X_train = std_scaler.fit_transform(X_train)


In [246]:
#fit and transform the test set 
X_test = std_scaler.fit_transform(X_test)


### Multiple Regression model using training dataset.

In [247]:
#fit the model
from sklearn.linear_model import LinearRegression
lr = LinearRegression()
lr = lr.fit(X_train,y_train)



In [248]:
#print intercept and coefficients for train model
i = lr.intercept_
c = lr.coef_
print('Intercept: \n', i)
print('Coefficients: \n', c)


Intercept: 
 [154.30311615]
Coefficients: 
 [[ -0.63205983 -10.66019076  25.27356488  15.40178564  48.95367067
  -47.94348653 -29.58476333   2.32732114   5.04980454]]


### predict the test set result

In [249]:
lr = lr.fit(X_test,y_test)
y_pred = lr.predict(X_test)


In [250]:
y_pred[0:10]

array([[159.02335107],
       [100.62703281],
       [129.32879923],
       [127.656188  ],
       [191.57009644],
       [ 49.8311031 ],
       [ 88.54967754],
       [156.46248758],
       [129.69106278],
       [114.58299047]])

###  r2_score on test data

In [252]:
#Accuracy on test data
from sklearn.metrics import r2_score

test_data_accuracy = r2_score(y_test, y_pred)

print('r2_score is :',test_data_accuracy)

r2_score is : 0.5349474663568761
